dotnet/src/Functions/Functions.OpenApi/Extensions/OpenApiKernelExtensions.cs

```diff
@@ -198,6 +225,124 @@ public static class OpenApiKernelExtensions
             cancellationToken: cancellationToken).ConfigureAwait(false);
     }
 
+    /// <summary>
+    /// Creates a kernel plugin from an API manifest file asynchronously.
+    /// </summary>
+    /// <param name="kernel">The kernel instance.</param>
+    /// <param name="pluginName">The name of the plugin.</param>
+    /// <param name="filePath">The file path of the API manifest.</param>
+    /// <param name="executionParameters">Optional execution parameters for the API functions.</param>
+    /// <param name="cancellationToken">Optional cancellation token.</param>
+    /// <returns>A task that represents the asynchronous operation. The task result contains the created kernel plugin.</returns>
+    [Experimental("SKEXP0099")]
+    public static async Task<KernelPlugin> CreatePluginFromApiManifestAsync(
+        this Kernel kernel,
+        string pluginName,
+        string filePath,
+        OpenApiFunctionExecutionParameters? executionParameters = null,
+        CancellationToken cancellationToken = default)
+    {
+        Verify.NotNull(kernel);
+        Verify.ValidPluginName(pluginName, kernel.Plugins);
+
+#pragma warning disable CA2000 // Dispose objects before losing scope. No need to dispose the Http client here. It can either be an internal client using NonDisposableHttpClientHandler or an external client managed by the calling code, which should handle its disposal.
+        var httpClient = HttpClientProvider.GetHttpClient(executionParameters?.HttpClient ?? kernel.Services.GetService<HttpClient>());
+#pragma warning restore CA2000
+
+        if (!File.Exists(filePath))
+        {
+            throw new FileNotFoundException($"ApiManifest file not found: {filePath}");
+        }
+
+        string apiManifestFileJsonContents = await DocumentLoader.LoadDocumentFromFilePathAsync(filePath,
+            kernel.LoggerFactory.CreateLogger(typeof(OpenApiKernelExtensions)) ?? NullLogger.Instance,
+            cancellationToken).ConfigureAwait(false);
+        JsonDocument jsonDocument = JsonDocument.Parse(apiManifestFileJsonContents);
+
+        ApiManifestDocument document = ApiManifestDocument.Load(jsonDocument.RootElement);
+
+        var functions = new List<KernelFunction>();
+        foreach (var apiDependency in document.ApiDependencies)
+        {
+            var apiName = apiDependency.Key;
+            var apiDependencyDetails = apiDependency.Value;
+
+            var apiDescriptionUrl = apiDependencyDetails.ApiDescriptionUrl;
+
+            var openApiDocumentString = await DocumentLoader.LoadDocumentFromUriAsync(new Uri(apiDescriptionUrl),
+                kernel.LoggerFactory.CreateLogger(typeof(OpenApiKernelExtensions)) ?? NullLogger.Instance,
+                httpClient,
+                authCallback: null,
+                HttpHeaderValues.UserAgent,
```

Comment:
The type name `OpenApiFunctionExecutionParameters` suggests that these are parameters to be used while calling the plugin functions (i.e. executing them). For example, we can't use the authCallback override from execution parameters in the line above because it is intended for the REST API call, not for fetching the OpenAPI document. Wouldn't the parameters be overloaded if used at the time of import? If the intention is to use for both, please let me know.